### PR TITLE
Check that fs errors have the proper kind

### PR DESCRIPTION
**Pull request by christianpoveda**
r? @RalfJung

### CODE REVIEW

**Comment by RalfJung** [comment]
Thanks!

@bors r+

**Comment by RalfJung** [comment]
A new compiletest-rs was released and needs a newer rustc (and we can't have a lockfile unfortunately)... I'm on it.

**Comment by RalfJung** [comment]
Now with a lockfile, this should works.

@bors r+

**Comment by RalfJung** [comment]
Looks like a legitimate test failure... Did these tests pass locally for you?

**Comment by christianpoveda** [comment]
oh sorry I had done those before you changed the test to use `&path` so I rebased without testing... my bad. Let me fix it

-------- Original Message --------

File: tests/run-pass/fs.rs [hunk 1]
```diff
@@ -23,8 +23,10 @@ fn main() {
     assert_eq!(bytes, contents.as_slice());
     // Removing file should succeed
     remove_file(&path).unwrap();
-    // Opening non-existing file should fail
-    assert!(File::open(&path).is_err());
-    // Removing non-existing file should fail
-    assert!(remove_file(&path).is_err());
+
+    // The two following tests also check that the `__errno_location()` shim is working properly.
+    // Opening non-existing file should fail with a not found error
```
Comment (RalfJung) [inline tests/run-pass/fs.rs]:
```suggestion
    // Opening non-existing file should fail with a not found error.
```

File: tests/run-pass/fs.rs [hunk 2]
```diff
@@ -23,8 +23,10 @@ fn main() {
...
+    assert_eq!(ErrorKind::NotFound, File::open(&path).unwrap_err().kind());
+    // Removing non-existing file should fail with a not found error
```
Comment (RalfJung) [inline tests/run-pass/fs.rs]:
```suggestion
    // Removing non-existing file should fail with a not found error.
```

**Pull request merged**